sources/api/migration/migration-helpers/src/common_migrations.rs

```diff
@@ -152,6 +153,206 @@ mod test_add_prefixes_migration {
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.data,
+            hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+            }
+        );
+    }
+
+    #[test]
+    fn compound_suffix() {
+        let data = MigrationData {
+            data: hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "remove.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+                "remove.not.this".into() => 0.into(),
+            },
+            metadata: HashMap::new(),
+        };
+        let result = AddPrefixSuffixMigration(vec![PrefixSuffix {
+            prefix: "remove",
+            suffix: "like.this",
+        }])
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.data,
+            hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+                "remove.not.this".into() => 0.into(),
+            }
+        );
+    }
+
+    #[test]
+    fn multiple_entries() {
+        let data = MigrationData {
+            data: hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "remove.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+                "remove.also.this".into() => 0.into(),
+                "delete.something.here".into() => 0.into(),
+            },
+            metadata: HashMap::new(),
+        };
+        let result = AddPrefixSuffixMigration(vec![
+            PrefixSuffix {
+                prefix: "remove",
+                suffix: "this",
+            },
+            PrefixSuffix {
+                prefix: "delete",
+                suffix: "here",
+            },
+        ])
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.data,
+            hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+            }
+        );
+    }
+
+    #[test]
+    fn no_match() {
+        let data = MigrationData {
+            data: hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+                "other.setting.here".into() => 0.into(),
+            },
+            metadata: HashMap::new(),
+        };
+        let result = AddPrefixSuffixMigration(vec![PrefixSuffix {
+            prefix: "remove.",
+            suffix: ".this",
+        }])
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.data,
+            hashmap! {
+                "keep.stuff.like.this".into() => 0.into(),
+                "keep.this.too".into() => 0.into(),
+                "other.setting.here".into() => 0.into(),
+            }
+        );
+    }
+
+    #[test]
+    fn tight_matching() {
```

Comment:
Added more test cases here with real settings - for the more strict regex matching.